### PR TITLE
Support detect VM level process

### PR DESCRIPTION
**Pull request by mrproliu**

### CODE REVIEW

File: configs/rover_configs.yaml [hunk 1]
```diff
@@ -32,4 +32,28 @@ core:
...
+process_discovery:
+  active: ${ROVER_PROCESS_DISCOVERY_ACTIVE:true}
+  # The period of report or keep alive process(second)
+  heartbeat_period: ${ROVER_PROCESS_DISCOVERY_HEARTBEAT_PERIOD:20}
+  # Detect processes in VM mode
+  vm:
+    # Is active the VM mode to detect processes
+    active: ${ROVER_PROCESS_DISCOVERY_VM_ACTIVE:true}
+    # The period to detect the process(second)
+    period: ${ROVER_PROCESS_DISCOVERY_VM_PERIOD:15}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:46]: This is a local mechanism, let's make it more accurate. Try ms as unit, and set 500ms as default.

File: configs/rover_configs.yaml [hunk 2]
```diff
@@ -32,4 +32,28 @@ core:
...
+    finders:
+      # Use regex string to locate the processes
+      # Duplicate entities cannot be reported. If multiple entity are generated, only one process will be report
+      # If the multiple finders could match the same one process, only the first finder could be selected and report
+      - match_cmd_regex: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_MATCH_CMD_REGEX:rover}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:51]:
```suggestion
      - match_cmd_regex: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_MATCH_CMD_REGEX:}
```
I think we should set default as empty string, and make Rover not be able to boot(without manual setting).

File: configs/rover_configs.yaml [hunk 3]
```diff
@@ -32,4 +32,28 @@ core:
     # How frequently to check the connection(second)
     check_period: ${ROVER_BACKEND_CHECK_PERIOD:5}
     # The auth value when send request
-    authentication: ${ROVER_BACKEND_AUTHENTICATION:""}
\ No newline at end of file
+    authentication: ${ROVER_BACKEND_AUTHENTICATION:""}
+
+process_discovery:
+  active: ${ROVER_PROCESS_DISCOVERY_ACTIVE:true}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:38]: I think this is not necessary, this must be activated?

File: configs/rover_configs.yaml [hunk 4]
```diff
@@ -32,4 +32,28 @@ core:
...
+  # The period of report or keep alive process(second)
+  heartbeat_period: ${ROVER_PROCESS_DISCOVERY_HEARTBEAT_PERIOD:20}
+  # Detect processes in VM mode
+  vm:
+    # Is active the VM mode to detect processes
+    active: ${ROVER_PROCESS_DISCOVERY_VM_ACTIVE:true}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:44]:
```suggestion
    active: ${ROVER_PROCESS_DISCOVERY_VM_ACTIVE:false}
```

As I asked for https://github.com/apache/skywalking-rover/pull/5/files#r815953896, it is better to disable this.

File: configs/rover_configs.yaml [hunk 5]
```diff
@@ -32,4 +32,28 @@ core:
...
+    period: ${ROVER_PROCESS_DISCOVERY_VM_PERIOD:15}
+    finders:
+      # Use regex string to locate the processes
+      # Duplicate entities cannot be reported. If multiple entity are generated, only one process will be report
+      # If the multiple finders could match the same one process, only the first finder could be selected and report
+      - match_cmd_regex: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_MATCH_CMD_REGEX:rover}
+        # The Layer need to relate to the process entity
+        layer: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_LAYER:VM}
+        # The Service Name need to relate to the process entity
+        service_name: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_SERVICE_NAME:rover}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:55]:
```suggestion
        service_name: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_SERVICE_NAME:}
```

Same as https://github.com/apache/skywalking-rover/pull/5/files#r815955089 ask, we should keep this blank, and blank should be illegal to boot.

File: configs/rover_configs.yaml [hunk 6]
```diff
@@ -32,4 +32,28 @@ core:
...
+        # The Service Instance Name need to relate to the process entity
+        instance_name: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_INSTANCE_NAME:{{.Rover.HostIP}}}
+        # The Process Name need to relate to the process entity
+        process_name: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_PROCESS_NAME:{{.Process.ExeName}}}
```
Comment (wu-sheng) [inline configs/rover_configs.yaml:59]: Please explain `ExeName` in the comment.

File: configs/rover_configs.yaml (hunk 1 above)
Comment (mrproliu) [reply configs/rover_configs.yaml]: Updated to use the time duration format, and make the default is `3s`, 500ms may too frequently.

File: configs/rover_configs.yaml (hunk 5 above)
Comment (wu-sheng) [reply configs/rover_configs.yaml:55]: Missing this?

File: configs/rover_configs.yaml (hunk 2 above)
Comment (mrproliu) [reply configs/rover_configs.yaml]: Updated. If the `match_cmd_regex` is not set, the Rover could not be booted.

File: configs/rover_configs.yaml (hunk 4 above)
Comment (mrproliu) [reply configs/rover_configs.yaml]: Have disabled by defualt.

File: configs/rover_configs.yaml [hunk 7]
```diff
@@ -32,4 +32,28 @@ core:
...
+      # Use regex string to locate the processes
+      # Duplicate entities cannot be reported. If multiple entity are generated, only one process will be report
+      # If the multiple finders could match the same one process, only the first finder could be selected and report
+      - match_cmd_regex: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_MATCH_CMD_REGEX:rover}
+        # The Layer need to relate to the process entity
+        layer: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_LAYER:VM}
+        # The Service Name need to relate to the process entity
+        service_name: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_SERVICE_NAME:rover}
+        # The Service Instance Name need to relate to the process entity
+        instance_name: ${ROVER_PROCESS_DISCOVERY_VM_FINDER_INSTANCE_NAME:{{.Rover.HostIP}}}
```
[the remark this answers is not in the document]
Comment (mrproliu) [reply configs/rover_configs.yaml]: Update to `{{.Rover.HostIPV4 "en0"}}` as default.

File: configs/rover_configs.yaml (hunk 5 above)
Comment (mrproliu) [reply configs/rover_configs.yaml]: Updated.

**Comment by wu-sheng** [comment]
Are you going to generate doc later?

File: pkg/process/finders/vm/finder.go [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+	vm2 := p2.(*Process)
+	return p1.Pid() == p2.Pid() && vm1.cmd == vm2.cmd && p1.Entity().SameWith(p2.Entity())
+}
+
+func (p *ProcessFinder) BuildEBPFProcess(ctx *base.BuildEBPFProcessContext, ps base.DetectedProcess) *v3.EBPFProcessProperties {
+	hostProcess := &v3.EBPFHostProcessMetadata{}
+	hostProcess.HostIP = ctx.HostIP
+	hostProcess.Pid = ps.Pid()
+	hostProcess.Cmd = ps.(*Process).cmd
+	hostProcess.Entity = &v3.EBPFProcessEntityMetadata{
```
Comment (wu-sheng) [inline pkg/process/finders/vm/finder.go:89]: Do we have any mechanism to verify whether this entity is duplicated? Or should we check on the server side?

Comment (mrproliu) [reply pkg/process/finders/vm/finder.go:89]: Yes, I have a mechanism to verify their entity is the same. Each VM process has [an identity](https://github.com/apache/skywalking-rover/pull/5/files#diff-ed74b624b51883d381869a88732798903eb04b05ec50f7b2435a25eb7cfc0a6eR64-R67), If the same identity could have multiple matched processes, then I would [only use the first one and log them](https://github.com/apache/skywalking-rover/pull/5/files#diff-9ac007c61d6cca27e5e4c59578707fa481fdb52a26f9fa26f8f841862b027607R210-R236).

**Comment by mrproliu** [comment]
> Are you going to generate doc later?

Yes, After finishing the profiling step, I would add the documentation.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**